src/main/java/org/opentripplanner/middleware/otp/OtpRequestProcessor.java

```diff
@@ -61,11 +77,18 @@ private static String proxy(Request request, spark.Response response) {
         }
         // Get request path intended for OTP API by removing the proxy endpoint (/otp).
         String otpRequestPath = request.uri().replace(OTP_PROXY_ENDPOINT, "");
-        // attempt to get response from OTP server based on requester's query parameters
-        OtpDispatcherResponse otpDispatcherResponse = OtpDispatcher.sendOtpRequest(request.queryString(), otpRequestPath);
-        if (otpDispatcherResponse == null || otpDispatcherResponse.responseBody == null) {
-            logMessageAndHalt(request, HttpStatus.INTERNAL_SERVER_ERROR_500, "No response from OTP server.");
-            return null;
+
+        OtpDispatcherResponse otpDispatcherResponse;
+        // If the request is to the plan end point and E2E is enabled mock the plan response
+        if (otpRequestPath.endsWith(OTP_PLAN_ENDPOINT) && runE2E) {
+            otpDispatcherResponse = getMockPlanResponse();
+        } else {
```

Comment:
I believe that in general it is a good practice to minimize the amount of test-specific code that is present in the regular code. It seems like the purpose of this is to return custom OTP responses and not rely on a live OTP server. A better approach might be to have the test code setup a local OTP instance at a certain port and then control what responses that server sends back.

Reply:
Maybe do something like this: https://stackoverflow.com/a/43176400/269834

Reply:
@evansiroky agree. I have implemented a second Spark server to mock the OTP for review.